### ISSUE TITLE
Feature Request: Offer validation of exported items in `__all__`

### ISSUE THREAD

**Issue by kkirsche** [astral-sh/ruff#6619]
This feature request is to ask if it would be possible to provide optional support for validating that the contents are `__all__` in an `__init__.py` file or equivalent are actually imported or cannot be determined due to a `from package import *` method.

For example:

```python
from package.module.client import A
from package.module.enums import B
from package.module.server import C

__all__ = ["A", "B", "C", "D"]
```

In this example, it would be great if ruff was able to warn the developer that `D` is not imported and because there are no `*` imports, it doesn't appear to exist

**Comment by edgarrmondragon** [comment]
I think this is covered by [F405](https://beta.ruff.rs/docs/rules/undefined-local-with-import-star-usage/) and [F822](https://beta.ruff.rs/docs/rules/undefined-export/).

**Comment by edgarrmondragon** [comment]
FWIW

```python
# myfile.py
from package.module.client import A
from package.module.enums import B
from package.module.server import C

__all__ = ["A", "B", "C", "D"]
```

```console
$ ruff check --isolated --select F myfile.py
iso.py:5:1: F822 Undefined name `D` in `__all__`
Found 1 error.
```

---

```python
# myfile.py
from package.module.client import A
from package.module.enums import B
from package.module.server import *

__all__ = ["A", "B", "C", "D"]
```

```console
$ ruff check --isolated --select F myfile.py
iso.py:3:1: F403 `from package.module.server import *` used; unable to detect undefined names
iso.py:5:1: F405 `C` may be undefined, or defined from star imports
iso.py:5:1: F405 `D` may be undefined, or defined from star imports
Found 3 errors.
```

**Comment by kkirsche** [comment]
I'll close this then and look into my config. Thank you and sorry for the overhead caused by this issue.
